datalad/core/distributed/push.py

```diff
@@ -123,16 +116,27 @@ class Push(Interface):
             data or changes for those paths are considered for a push.""",
             nargs='*',
             constraints=EnsureStr() | EnsureNone()),
+        data=Parameter(
+            args=("--data",),
+            doc="""what to do with data (annex'ed) data. 'anything' would cause
+            transfer of all annexed content, 'nothing' would avoid call to
+            `git annex copy` altogether. 'auto' would use 'git annex copy' with
```

Comment:
Should we drop 'auto' until someone requests it?  It seems no one complained about `publish` not honoring numcopies, so perhaps 'auto-if-wanted' is sufficient.

Reply:
If we do that, what would remain is identical to `datalad datalad.push.copy-auto-if-wanted=true push` -- according to https://github.com/datalad/datalad/pull/4620#issuecomment-641833047

I started using this code in daily operations, will report.

Reply:
`auto` mode is to be able (not yet tested) to make `publish` to use `push` since it is it's mode of operation - to not copy data by default, unless wanted is set, and no paths are explicitly specified